src/absil/ilwrite.fs

```diff
@@ -3916,15 +3936,7 @@ let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer:
 
     let timestamp = absilWriteGetTimeStamp ()
 
-    let os = 
-        try  
-            new BinaryWriter(FileSystem.FileStreamCreateShim(outfile))
-        with e -> 
-            failwith ("Could not open file for writing (binary mode): " + outfile)    
-
-    let  pdbData,debugDirectoryChunk,debugDataChunk,textV2P,mappings =
-        try 
-      
+    let gen os (pdbfile: bool) =
```

Comment:
Is it possible to avoid having a new inner function here?  It is only called in one place.   Why not put the prologue that computes "os" and "pdbfile" in the portion directly above this?  I'm looking to minimize the diff and keep thee current overall function structure the same